lib/rubocop/ast/traversal.rb

```diff
@@ -49,16 +51,23 @@ def on_#{type}(node)        # def on_send(node)
           end
         end
 
-        def body(signature, prelude)
-          signature
-            .map.with_index do |arg, i|
-              TEMPLATE[arg].gsub('%<index>i', i.to_s)
+        def body(child_node_types, expected_children_count)
+          debug = ENV.fetch('RUBOCOP_DEBUG', false)
+          children_check_code = debug ? children_check(expected_children_count) : ''
+          visit_children_code =
+            child_node_types
+            .map.with_index do |child_type, i|
+              TEMPLATE.fetch(child_type).gsub('%<index>i', i.to_s)
             end
-            .unshift(prelude)
             .join("\n")
+
+          <<~BODY
+            #{children_check_code}
+            #{visit_children_code}
+          BODY
         end
 
-        def arity_check(range)
+        def children_check(range)
```

Comment:
I'd prefer to be more precise here: `children_count_check` or `check_children_count`

Reply:
I renamed to `children_count_check_code` and moved the env variable handling there too so I don't have to think of another name for the local variable that used to hold the code